addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -2545,7 +2545,9 @@ export class Wysiwyg extends Component {
                         this.toggleLinkTools({forceDialog: true});
                         // Force the button style after the link modal is open.
                         setTimeout(() => {
-                            $(".o_link_dialog .link-style[value=primary]").click();
+                            const selectEl = document.querySelector('.o_link_dialog .form-select');
+                            selectEl.value = 'primary';
+                            selectEl.dispatchEvent(new Event('change'));
```

Comment:
Why is this `dispatchEvent` needed? Whatever the reason, it needs a comment above this line to explain it.

Reply:
Unlike click event, changing value of `select` element using javascript doesn't trigger the `change` event. So here we need to dispatch the `change` event forcefully. But yes, we need a piece of comment here :)

Reply:
Done changes :+1: .

Reply:
I understand that it doesn't trigger the change event. I meant _why_ the change event is needed ? Is some code bound on it ?

Reply:
Yes, in `link_dialog.js` we have a listener `_onTypeChange` which is responsible for collapsing.

Reply:
I see, thanks! Could you update the comment to the following please?
```js
                            // Dynamically changing value of select option
                            // doesn't trigger a change event. Trigger
                            // listeners by dispatching the event manually.
```

Reply:
Changes done :+1:

Reply:
Thanks!